kubectl-cloudflow/config/config.go

```diff
@@ -372,17 +374,97 @@ func validateLabelValue(labelValue string, label string) error {
+		for key, value := range volumesConfig.Root().GetObject().Items() {
+			secret := value.GetObject().GetKey("secret")
+			if secret == nil || !secret.IsObject() {
+				return fmt.Errorf("missing or malformed 'secret' in volume %s. Please have a look at documentation to see what's expected", key)
+			}
+			name := secret.GetObject().GetKey("name")
+			if name == nil {
+				return fmt.Errorf("missing 'name' in %s.secret", key)
+			}
+			if name.IsEmpty() {
+				return fmt.Errorf("missing content of 'name' in %s.secret.name", key)
+			}
+		}
+	}
+	return nil
+}
+
+func validateVolumesMounts(containersConfig *configuration.Config) error {
+	allowedProperties := []string{"mountPath", "readOnly", "subPath"}
+	for containerName := range containersConfig.Root().GetObject().Items() {
+		if containerConfig := containersConfig.GetConfig(containerName); containerConfig != nil && containerConfig.Root().IsObject() {
+			if volumesMountsConfig := containerConfig.GetConfig(volumeMountsKey); volumesMountsConfig != nil && volumesMountsConfig.Root().IsObject() {
+				for volumeMountName, volumeMountKeyValue := range volumesMountsConfig.Root().GetObject().Items() {
+					for key, value := range volumeMountKeyValue.GetObject().Items() {
+						if !contains(allowedProperties, key) {
+							return fmt.Errorf("not allowed '%s' key in the volume-mounts '%s'. Properties allowed are: %s ", key, volumeMountName, allowedProperties)
+						}
+						if value.IsEmpty() {
+							return fmt.Errorf("key '%s' has not value in then volume-mounts '%s' .", key, volumeMountName)
+						}
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func checkVolumeMountsReferToVolume(podsConfig *configuration.Config, containersConfig *configuration.Config) error {
+
+	validateVolumes(podsConfig.GetConfig("pod"))
+	validateVolumesMounts(containersConfig)
+
+	var volumesNames []string
+	var volumeMountSecretNames []string
+	if volumesConfig := podsConfig.GetConfig("pod").GetConfig(volumes); volumesConfig != nil && volumesConfig.Root().IsObject() {
+		for volumeName, _ := range volumesConfig.Root().GetObject().Items() {
+			volumesNames = append(volumesNames, volumeName)
+		}
+	}
+	for containerName := range containersConfig.Root().GetObject().Items() {
+		if containerConfig := containersConfig.GetConfig(containerName); containerConfig != nil && containerConfig.Root().IsObject() {
+			if volumesMountsConfig := containerConfig.GetConfig(volumeMountsKey); volumesMountsConfig != nil && volumesMountsConfig.Root().IsObject() {
+				for volumeMountName, _ := range volumesMountsConfig.Root().GetObject().Items() {
+					if !contains(volumesNames, volumeMountName) {
+						return fmt.Errorf("the volume-mounts '%s' should match with a volume.secret.name that is among '%s'", volumeMountName, volumesNames)
+					}
+					volumeMountSecretNames = append(volumeMountSecretNames, volumeMountName)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func contains(s []string, e string) bool {
```

Comment:
This aspect of Golang is just so sad.....

Reply:
That's exactly what I thought @olofwalker . It took me a few blogs to believe this is the case

Reply:
It's as if they don't even want you to write funcs like these...

Reply:
hehe